src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -1,87 +1,96 @@
-    public static final BigInteger DIFFICULTY_CAP_TESTNET = BigInteger.valueOf(1000000000000000L);
-    public static final BigInteger DIFFICULTY_CAP_REGTEST = BigInteger.valueOf(20);
-    private final BigInteger difficultyCap;
-    private BigInteger difficultyTarget = BigInteger.valueOf(3);
-    private int maxAmountBlockHeaders = 7;
-    private long informerInterval = 2_000;
-    private boolean stopBookkeepingScheduler = false;
-    private int maxChunkSizeToHsm = 10;
+    public enum NetworkDifficultyCap {
+        MAINNET(new BigInteger("7000000000000000000000")),
+        TESTNET(BigInteger.valueOf(1000000000000000L)),
+        REGTEST(BigInteger.valueOf(20L));
 
-    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
-        if (signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget")) {
-            this.difficultyTarget = new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"));
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxAmountBlockHeaders")) {
-            this.maxAmountBlockHeaders = signerConfig.getConfig().getInt("bookkeeping.maxAmountBlockHeaders");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.informerInterval")) {
-            this.informerInterval = signerConfig.getConfig().getLong("bookkeeping.informerInterval");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.stopBookkeepingScheduler")) {
-            this.stopBookkeepingScheduler = signerConfig.getConfig().getBoolean("bookkeeping.stopBookkeepingScheduler");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxChunkSizeToHsm")) {
-            this.maxChunkSizeToHsm = signerConfig.getConfig().getInt("bookkeeping.maxChunkSizeToHsm");
+        private final BigInteger difficultyCap;
+
+        NetworkDifficultyCap(BigInteger difficultyCap) {
+            this.difficultyCap = difficultyCap;
         }
 
-        switch (networkParameter) {
-            case NetworkParameters.ID_MAINNET:
-                this.difficultyCap = DIFFICULTY_CAP_MAINNET;
-                break;
-            case NetworkParameters.ID_TESTNET:
-                this.difficultyCap = DIFFICULTY_CAP_TESTNET;
-                break;
-            case NetworkParameters.ID_REGTEST:
-                this.difficultyCap = DIFFICULTY_CAP_REGTEST;
-                break;
-            default:
-                String message = "Invalid network specified for the Bookkeeping Config: " + networkParameter;
-                logger.error(message);
-                throw new IllegalArgumentException(message);
+        public BigInteger getDifficultyCap() {
+            return difficultyCap;
         }
     }
 
+    private static final Logger logger = LoggerFactory.getLogger(PowHSMBookkeepingConfig.class);
+
+    private static final String DIFFICULTY_TARGET_PATH = "bookkeeping.difficultyTarget";
```

Comment:
Maybe for another story, but I'd consider moving the paths to an enum on a separate file. To clean up this class a bit.

Same for NetworkDifficultyCap enum, in a separate file it will simplify this class.

Default values make sense to keep them here I think

Reply:
I think we can make this in another story, so we can also put the paths present in `src/main/java/co/rsk/federate/config/FedNodeSystemProperties.java` in this enum as well (this way we consolidate a bit). We can come up with a better overall design and try to see how we can keep the defaults there. We can also include NetworkDifficultyCap there. I can create this story if you want.